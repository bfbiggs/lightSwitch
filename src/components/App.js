import PropTypes from 'prop-types';
import React from 'react';
import { hot } from 'react-hot-loader';
import { Route, Switch } from 'react-router-dom';
import HomePage from './Home';

export const LightContext = React.createContext();

function App() {
  const [status, toggleLight] = React.useState(false);

  return (
    <LightContext.Provider value={{
      status,
      toggleFn: () => toggleLight(!status)
    }}>
      <div className={
        "md-main" +
        `${status ? '' : ' md--dark'} `
      }>
        <Switch>
          <Route exact path="/" component={HomePage} />
        </Switch>
      </div>
    </LightContext.Provider>
  );
}

App.propTypes = {
  children: PropTypes.element,
};

export default hot(module)(App);
