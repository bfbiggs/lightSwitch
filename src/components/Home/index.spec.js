import React from 'react';
import { mount, shallow } from 'enzyme';
import HomePage from './index';
import { LightContext } from '../App';

describe('<HomePage />', () => {
  it('should match SnapShot', () => {
    const container = shallow(
      <LightContext.Provider value={{
        status: false,
        toggleFn: () => {}
      }}>
        <HomePage />
      </LightContext.Provider>
      );
    expect(container).toMatchSnapshot();
  });

  it('should handle onChange event', () => {
    const addItem = jest.fn();

    const container = mount(
      <LightContext.Provider value={{
        status: false,
        toggleFn: () => addItem()
      }}>
        <HomePage />
      </LightContext.Provider>
    );

    container.find('input').simulate('change');
    expect(addItem).toHaveBeenCalled();
  });


});
