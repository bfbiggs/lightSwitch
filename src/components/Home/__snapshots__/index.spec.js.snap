// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should match SnapShot 1`] = `ShallowWrapper {}`;
